File: ruby/grains/grains.rb
module Grains

  NUMBER_OF_SQUARES = 64

  def self.square(number)
    # Validate number input
    unless (1..64).include?(number)
      raise ArgumentError.new
    end
    # Apply geometric series formula
    # as a bitwise operation
    1 << (number-1)
  end

  def self.total
  # The constant could easily be a variable,
  # if we wanted the total at any given point
    (1 << NUMBER_OF_SQUARES) - 1
  end
  
end

File: python/grains/grains.py
def square(number):
    if number <= 0 or number > 64:
        raise ValueError("square must be between 1 and 64")
    squares = [1]
    for i in range(number-1):
        squares.append(squares[-1]*2)
    return squares[-1]

def total():
    total = 1
    for i in range(2, 65):
        total += square(i)
    return total
